src/detection_win.cpp

```diff
@@ -423,6 +424,21 @@ void ExtractDeviceInfo(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, TCHAR
 		// Use this to extract VID / PID
 		extractVidPid(buf, resultItem);
 	}
+
+	DWORD dwCap = 0x0;
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_CAPABILITIES, &DataT, (PBYTE)&dwCap, 4, &nSize)) {
+		// If device has a CM_DEVCAP_UNIQUEID capability, then we assume it has a serial number
+		if ((dwCap & CM_DEVCAP_UNIQUEID) == CM_DEVCAP_UNIQUEID) {
+			if (DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, buffSize, &nSize)) {
+				// extract Serial Number
+				string str = buf;
```

Comment:
Let's rename this variable to `string deviceInstanceId = buf;` so it better represents what it holds

Reply:
I am OK with that